client/me/purchases/cancel-purchase/button.jsx

```diff
@@ -26,6 +31,70 @@ const CancelPurchaseButton = React.createClass( {
 		};
 	},
 
+	handleCancelPurchaseClick() {
+		if ( isDomainRegistration( this.props.purchase ) ) {
+			return this.goToCancelConfirmation();
+		}
+
+		this.setState( {
+			showDialog: true
+		} );
+	},
+
+	closeDialog( action ) {
+		if ( action === 'delete' ) {
+			this.setState( {
+				submitting: true
+			} );
+
+			return this.submitCancelPurchase();
+		}
+
+		this.setState( {
+			showDialog: false
+		} );
+	},
+
+	renderCancelConfirmationDialog() {
+		const { domain, priceText } = this.props.purchase,
+			purchaseName = getName( this.props.purchase ),
+			buttons = [
+				{ action: 'cancel', label: this.translate( "No, I'll Keep It" ) },
```

Comment:
Super minor, but it feels like the button that is labeled "Yes, Cancel Now" is the one that should have the `action: 'cancel'`.